mcpgateway/plugins/framework/models.py

```diff
@@ -662,17 +668,32 @@ class GlobalContext(BaseModel):
     user: Optional[str] = None
     tenant_id: Optional[str] = None
     server_id: Optional[str] = None
+    state: dict[str, Any] = {}
```

Comment:
I think having mutable defaults (`{}`) assign this way can be an issue; the dict is created once at class definition time and may be shared across instances. Use `Field(default_factory=dict)` instead. This could cause subtle state bleed across requests/plugins (didn't verify it).

```python
state: dict[str, Any] = {}
metadata: dict[str, Any] = {}
```

Replace `= {}` with `Field(default_factory=dict)` for all dict fields (state, metadata), and consider the same for any lists/sets.